CHANGES.rst

```diff
@@ -10,11 +10,38 @@ Changes
 
 Changes:
 --------
-- No change.
+- Refactor Workflow operation flow to reuse shared input and output staging operations between implementations.
+  Each new step process implementation now only requires to implement the specific operations related to deployment,
+  execution, monitoring and result retrieval for their process, without need to consider Workflow intermediate staging
+  operations to transfer files between steps.
+- Refactor ``Wps1Process`` and ``Wps3Process`` step processes to follow new workflow operation flow.
```

Comment:
One day we will need to rename that wps3 reference...

Reply:
With the current trend of everything becoming "OGC API - <type>", I don't think we need to worry about it too much. 
WPS is also (unfortunately) one of the least maintained standard in the OWS suite. I highly doubt a XML WPS-3 successor will ever emerge.

Reply:
My point was more to eventually replace that naming in favor of the more appropriate `OGC API - PROCESS` since wps3 refers to nothing.

Reply:
I left it as Wps3 because it technically implements more than just OGC API - Processes. 
It handles Docker step and Workflow at the same time.
https://github.com/crim-ca/weaver/pull/372/files#diff-ce41af74805c1d891a393bd1488434a450041ca7cc7df4f3532b4e516badf656R1714-R1721